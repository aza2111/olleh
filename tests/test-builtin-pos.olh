//tests builtin functionalities

print dictionary.getLength();
readDict("example.txt");
/*example.txt contains:
anagram
function
function
please*/

print dictionary.getLength();
foreach w dictionary:
        print w + ": " + dictionary.get(w);
end;

print anagram("asdkf");
print anagram("margana");

print dictionary.contains("anagram");

map<char> mp = fresh map(char);
print mp.contains('x');
mp.destroy('x');
mp.set('5', 5);
print mp;
mp.destroy('5');
print mp;

dictionary.destroy("anagram");
print dictionary;

String xx = intToString(45);
print xx;

String banana = "banana";
print banana;
list<char> bananalist = stringToList(banana);
print bananalist;
banana = listToString(bananalist);
print banana;
