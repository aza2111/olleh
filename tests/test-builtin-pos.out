0
3
anagram: 1
function: 1
please: 1
(null)
anagram
true
false
{'5' -> 5}
{}
{"function" -> 1, "please" -> 1}
45
banana
['b', 'a', 'n', 'a', 'n', 'a']
banana
